public/index.php

```diff
@@ -4,6 +4,26 @@
  * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
  */
 
+if (ini_get('display_errors')) {
+    /*
+     * PHP by default returns HTTP 200 even when exceptions are thrown.
+     *
+     * Now by default return HTTP 500 so any API client can trust in the HTTP response code.
+     */
+    http_response_code(500);
```

Comment:
@Maks3w I think this error management should not be added in `index.php` of the skeleton application. I propose to add this exception handler in `zf-apigility` module. The goal, as you suggested, is to return a 500 error only for API. Moreover, I think we should add it only in production environment (we can check if the `/config/development.config.php` file does not exist). As @weierophinney said if we change the HTTP status for PHP errors we can have issue in debugging (using for instance XDebug).